parser_library/src/workspaces/workspace.cpp

```diff
@@ -947,17 +950,17 @@ std::vector<std::pair<std::string, size_t>> workspace::make_opcode_suggestion(
     std::vector<std::pair<std::string, size_t>> result;
 
     asm_option opts;
-    if (const auto* pgm = m_configuration.get_program(file); pgm)
-    {
-        auto& proc_grp = m_configuration.get_proc_grp_by_program(*pgm);
-        proc_grp.apply_options_to(opts);
-        pgm->asm_opts.apply_options_to(opts);
 
-        result = proc_grp.suggest(opcode, extended);
-    }
+    if (auto pgm = m_configuration.get_program(file); !pgm)
+        implicit_proc_grp.apply_options_to(opts);
+    else if (auto proc_grp = m_configuration.get_proc_grp_by_program(*pgm); !proc_grp)
+        implicit_proc_grp.apply_options_to(opts);
     else
     {
-        implicit_proc_grp.apply_options_to(opts);
+        proc_grp->apply_options_to(opts);
+        pgm->asm_opts.apply_options_to(opts);
+
```

Comment:
maybe extract to `bool workspace::apply_program_options(asm_option&)` returning true if the non-implicit group was found?
Also if the `pgm` is valid it should probably be used even if the group is unavailable?

Reply:
Was the new templated function `apply_options_to` something similar to what you've had in mind?